cmd/maya-apiserver/cstor-operator/spc/controller.go

```diff
@@ -149,13 +149,17 @@ func (cb *ControllerBuilder) Build() (*Controller, error) {
 	return cb.Controller, nil
 }
 
-// addSpc is the add event handler for spc.
+// addSpc is the add event handler for spc
 func (c *Controller) addSpc(obj interface{}) {
 	spc, ok := obj.(*apis.StoragePoolClaim)
 	if !ok {
 		runtime.HandleError(fmt.Errorf("Couldn't get spc object %#v", obj))
 		return
 	}
+	if spc.Labels[string(apis.OpenEBSUpgradeKey)] == "true" {
```

Comment:
This logic should be handled at informer handle functions i.e. AddFunc, UpdateFunc, DeleteFunc

Reply:
Informer is making call to these functions when ever it receives request